feat(log): add serilog basic implementation

Authfix opened the pull request.

Authfix committed:
feat(log): add serilog package

Add serilog package with a default sink (console)

Authfix committed:
feat(log): add serilog configuration support package

Authfix committed:
feat(log): implement serilog basic configuration

danielpetisme commented on src/JHipsterNetSampleApplication/Program.cs:
```diff
@@ -1,17 +1,75 @@
+        /// <returns>Return the current <see cref="IConfiguration"/></returns>
+        private static IConfiguration GetAppConfiguration()
+        {
+            var environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
```
I thought it was ASP.Net Core default behavior ?
Same for the `appsettings.json`, `appsettings.{environment}.json`.
Do we have any benefit to explicit that part rather than relying on the default config.?

Authfix replied:
In main entrypoint, we are not in an Asp.Net context by default but a simple C# Application context. Because of this, we don't have access to configuration to setup Serilog (it's `CreateDefaultBuilder` who add Configuration). 

We must create an `IConfiguration` instance before asp.net bootstrap if we want to use serilog at the hightest possible level.

Your snippet is good, it use a callback to give you access to the asp.net context when It will be setup. But it will prevent your application to catch exceptions with Serilog during the bootstrap process.

My snippet put Serilog at the entrypoint level to catch all possible exceptions. But we need to do extra work at startup.

So it depend if we want to catch startup logs with Serilog or not. We can argue that startup exceptions are special cases and we don't want to give Serilog responsability of those exceptions.

danielpetisme replied:
The main issue I have with a more "global" serilog bootstrap, is that it doesn't capture the `SourceContext` (since AspNet Core is not yet starter)....

Anyway, the PR is good, let's move on to the configuration of the logger

danielpetisme reviewed (commented):
Actually I've started the integration too (bad communication)
This is the snippet I have to configure the logger, it looks enought to configure Serilog
```
private static IWebHostBuilder CreateWebHostBuilder(params string[] args)
        {
            return WebHost.CreateDefaultBuilder(args)
                .UseSerilog((ctx, config) => {
                    config.Enrich.WithThreadId();
                    config.ReadFrom.Configuration(ctx.Configuration);
                })
                .UseStartup<Startup>();
        }
```

danielpetisme committed:
Basic configuration proposal

danielpetisme reviewed (approved):
The main issue for me with a global bootstrap is that you can't capture a `SourceContext` since AspNet Core is not yet starter.
Anyway, looks good enough to move on to the configuration of the logger itself.

Authfix committed:
feat(serilog): add comments and environment variable support for configuration

Authfix committed:
feat(serilog): update configuration

Update configuration to manage development (with debug logs) and production (with information logs) profiles

Authfix committed:
Merge branch 'feature/serilog' into danielpetisme 'feature/serilog'

Authfix committed:
Merge branch feature/serilog from danielpetisme

Add LoggerNameEnricher to serilog

danielpetisme commented:
:wave: @Authfix do you plan to add more stuff to the PR or we should merge and move to another topic?

danielpetisme commented:
Looks good enough for a first merge.
